frontend/appflowy_flutter/lib/plugins/document/document.dart

```diff
@@ -119,6 +121,15 @@ class DocumentPluginWidgetBuilder extends PluginWidgetBuilder
           view: view,
         ),
         const HSpace(4),
+        BlocProvider(
+          create: (context) =>
+              FavoriteBloc()..add(const FavoriteEvent.initial()),
+          child: DocumentFavoriteButton(
+            key: ValueKey('favorite_button_${view.id}'),
+            view: view,
+          ),
+        ),
+        const HSpace(4),
```

Comment:
rest of the logic seems fine, I believe this we should have a common instance of FavoriteBloc, try creating a single global instance of `BlocProvider<FavoriteBloc>` a bit higher in widget tree and remove redundant ones in `document.dart` and `sidebar.dart` since we should have a common state of favorites throughout the application.